File: C++/231_PowerofTwo.cpp
// Question - https://leetcode.com/problems/power-of-two/

#include <bits/stdc++.h>
using namespace std;
class Solution
{
public:
    bool isPowerOfTwo(int n)
    {
        if (n <= 0)
        {
            return false;
        }
        while (n > 1)
        {
            if (n % 2 != 0)
            {
                return false;
            }
            else
            {
                n /= 2;
            }
        }
        return true;
    }
};
int main()
{
    int t;
    cin >> t;
    Solution obj;
    while (t--)
    {
        int n;
        cin >> n;
        cout << (obj.isPowerOfTwo(n) ? "true" : "false") << endl;
    }

    return 0;
}

/*
class Solution {
public:
    bool isPowerOfTwo(int n) {
        if (n <= 0) {
            return false;
        }
        while (n > 1) {
            if (n % 2 != 0) {
                return false;
            } else {
                n /= 2;
            }

        }
         return true;
    }
};
*/

File: Java/231_poweroftwo.java
class Solution {
    public boolean isPowerOfTwo(int n) {
        if (n <= 0) {
            return false;
        }
        return Integer.bitCount(n) == 1; // Correct method name
    }

    public static void main(String[] args) {
        Solution obj = new Solution();
        int[] testCases = {1, 2, 4, 8, 16, 3, 5, 10};

        for (int n : testCases) {
            System.out.println(n + " is power of two: " + obj.isPowerOfTwo(n));
        }
    }
}



/*class Solution {
    public boolean isPowerOfTwo(int n) {
        if (n <= 0) {
            return false;
        }
       return Integer.bitCount(n) == 1;
        

    }
}*/